File: app/XDM/XDM.App.Core/HttpServer/LineReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using TraceLog;

namespace HttpServer
{
    internal static class LineReader
    {
        internal static IEnumerable<string> ReadLines(Stream stream)
        {
            //try
            //{
            // var lines = new List<string>();
            var buffer = new StringBuilder();
            while (true)
            {
                var x = stream.ReadByte();
                if (x == -1) throw new IOException("Unexpected EOF");
                if (x == '\n')
                {
                    if (buffer.Length == 0) yield break;
                    var line = buffer.ToString();
                    buffer = new StringBuilder();
                    yield return line;
                }
                else if (x != '\r') buffer.Append((char)x);
            }
            //return lines;
            //}
            //catch (Exception e)
            //{
            //    Log.Debug(e, "Error parsing headers");
            //    throw e;
            //}
        }
    }
}

File: xdm-app/src/main/java/xdm/integration/LineReader.kt
package xdm.integration

import java.io.IOException
import java.io.InputStream

object LineReader {
    fun readLine(source: InputStream): String {
        val buffer = StringBuilder()
        while (true) {
            val x: Int = source.read()
            if (x == -1) throw IOException("Unexpected EOF while reading header line")
            if (x == '\n'.code) return buffer.toString()
            if (x != '\r'.code) buffer.append(x.toChar())
        }
    }
}
